http_test.go

```diff
@@ -86,7 +90,34 @@ func TestIteratingByWorker(t *testing.T) {
 	}, testChunks, "should have sorted the metrics by hashes")
 }
 
-func TestServerImport(t *testing.T) {
+func testServerImport(t *testing.T, filename string, contentEncoding string) {
+
+	f, err := os.Open(filename)
+	assert.NoError(t, err, "Error reading response fixture")
+	defer f.Close()
+
+	r := httptest.NewRequest(http.MethodPost, "/import", f)
+	r.Header.Set("Content-Encoding", contentEncoding)
+
+	w := httptest.NewRecorder()
+
+	config := localConfig()
+	s := setupLocalServer(t, config)
+
+	handler := handleImport(&s)
+	handler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusAccepted, w.Code, "Test server returned wrong HTTP response code")
+}
+
+func TestServerImportCompressed(t *testing.T) {
+	// Test that the global veneur instance can handle
+	// requests that provide compressed metrics
+	testServerImport(t, filepath.Join("fixtures", "import.deflate"), "deflate")
```

Comment:
We're actually guaranteed that the test is invoked from the root directory, so that's not imposing any extra assumption for us. Finding the file at runtime is only a challenge *if* we use `go test -c` *and* move that binary somewhere else before running it. I don't see us doing that in our CI environment anytime soon, and since there are ways to solve that use case, we can cross that bridge if we come to it.


There are a lot of advantages of keeping the recorded responses in a plain text file, but the key thing that we care about is that we're actually *not* treating the data as an endogenous Go value. Among other things, this means that we are able to inspect and process the data without having to parse a Go sourcefile and walk the AST.